### ISSUE TITLE
Set default serialization arguments

### ISSUE THREAD

**Issue by Slyces** [pydantic/pydantic#10141]
### Initial Checks

- [X] I have searched Google & GitHub for similar requests and couldn't find anything
- [X] I have read and followed [the docs](https://docs.pydantic.dev) and still think this feature is missing

### Description

# Short Summary

I think providing a way to set the default arguments for serialisation methods (`model_dump, `model_dump_json`) would make multiple use-cases more robust.

# Context

Pydantic's `model_dump` and `model_dump_json` provide many valuable arguments when serialising.

All of those keywords arguments are here to enable a behaviour that you might or might not need. However, in my experience, **if I need this behaviour once, I need it (almost) everywhere**. I find that things like `by_alias`, `exclude_none`, ... are often tightly coupled with the design and constraints of your model through your entire project/application.

This leads me to be worried that I might have a complicated bug to spot if I use, let's say, `by_alias=True` everywhere, but forget it in one occurence.

When dealing with one model, it's possible to keep track. But in project where, let's say, your API should always hide default values in responses, the explicit `exclude_defaults=True` call can add up over every single one of your models.

_Note_: not all arguments make sense for this, for instance `exclude` can be set on the fields themselves

# Solution

My favourite solution in this situation would be to indicate the default behaviour on the model itself. I still have the flexibility to decide on a call by call basis, but I'm confident that if I (or another developper) just use `model_dump()`, it will follow the behaviour that fits my model's design and requirements.

## Example In House Implementation

To be fair, this is completely possible to do without upstream changes. Here's an example of code for a specific argument

```python
class CamelCaseModel(pydantic.BaseModel):
    model_config = pydantic.ConfigDict(alias_generator=to_camel,  populate_by_name=True)

    def model_dump(self, **kwargs: Any) -> dict[str, Any]:
        kwargs.setdefault("by_alias", True)
        return super().model_dump(**kwargs)

    def model_dump_json(self, **kwargs: Any) -> str:
        kwargs.setdefault("by_alias", True)
        return super().model_dump_json(**kwargs)
```
Here, `by_alias` will be set on every call, except if I know what I'm doing and provide an explicit value.

## Example Interface

If a solution was supported upstream, we could just use an interface like this one:
```python
class CamelCaseModel(pydantic.BaseModel):
    model_config = {
        "alias_generator": to_camel, 
        "populate_by_name": True,
        "dump_defaults": { "by_alias": True }  # Maybe "serialization_defaults"
    }
```

## Do we need this upstream?

I think the reason I would like this in pydantic is
- I think it's very useful and would make the API nicer
- Serialization is a key element of pydantic, while I find  
- I don't like having to edit they key methods in a shared base class
  - There can be potential issues with multiple inheritance (e.g. `ExcludeNoneModel` & `CamelCaseModel`)
  - It hides completion
- Some features like `exclude` already have an interface to be defined on model definition, showing that there's sometimes a need (at least for the `exclude` argument) to ensure a consistent behaviour on all serializations

I think this would also be a (nice) solution to https://github.com/pydantic/pydantic/issues/8551, which as mentioned in https://github.com/pydantic/pydantic/issues/8551#issuecomment-1892571228 is solved by **consistently** using `by_alias` 

### Affected Components

- [ ] [Compatibility between releases](https://docs.pydantic.dev/changelog/)
- [ ] [Data validation/parsing](https://docs.pydantic.dev/concepts/models/#basic-model-usage)
- [X] [Data serialization](https://docs.pydantic.dev/concepts/serialization/) - `.model_dump()` and `.model_dump_json()`
- [ ] [JSON Schema](https://docs.pydantic.dev/concepts/json_schema/)
- [ ] [Dataclasses](https://docs.pydantic.dev/concepts/dataclasses/)
- [ ] [Model Config](https://docs.pydantic.dev/concepts/config/)
- [ ] [Field Types](https://docs.pydantic.dev/api/types/) - adding or changing a particular data type
- [ ] [Function validation decorator](https://docs.pydantic.dev/concepts/validation_decorator/)
- [ ] [Generic Models](https://docs.pydantic.dev/concepts/models/#generic-models)
- [ ] [Other Model behaviour](https://docs.pydantic.dev/concepts/models/) - `model_construct()`, pickling, private attributes, ORM mode
- [ ] [Plugins](https://docs.pydantic.dev/) and integration with other tools - mypy, FastAPI, python-devtools, Hypothesis, VS Code, PyCharm, etc.

**Comment by Viicos** [comment]
Thanks for the well written feature request. While I understand the added value of having defaults set at the `model_config` level (this makes me think of the [layered architecture](https://docs.litestar.dev/2/usage/applications.html#layered-architecture) available in Litestar), I have one concern (as you mentioned with `by_alias`) related to the signature of the `model_dump(_json)` functions, that would become:

```python
def model_dump(
    self,
    *,
    ...,
    by_alias: bool | None = None,  # was `: bool = False` before. Now there's no way to easily remember the default value from the signature.
    ...,
) -> dict[str, Any]:
```

Due to the current implementation (serialization is on the `pydantic-core`) I also expect some work for this to happen. My take is we should probably wait for some upvotes before implementing this, considering the possibly added complexity and maintenance.

**Comment by sydney-runkle** [comment]
>  I also expect some work for this to happen. My take is we should probably wait for some upvotes before implementing this, considering the possibly added complexity and maintenance.

I agree with @Viicos here - given that there is a way for you to set default serialization settings by overriding `model_dump` or similar functions, I think this will be low priority given the additional maintenance burden it introduces.

That being said, I think there are some common serialization settings, like `by_alias`, that we plan to introduce explicit config settings for in v3, given demand, see https://github.com/pydantic/pydantic/issues/8379.

Thanks for your well thought out feature request - we very much appreciate the detail and organization.
